src/FsPickler/Format/BinaryFormat.fs

```diff
@@ -43,7 +43,7 @@ module private BinaryFormatUtils =
     [<Literal>]
     let flagMask  = 0x000000ffu
 
-    // this binary format uses Buffer.BlockCopy for performance
+    // this binary format uses Marshal.Copy for performance
```

Comment:
Why would we need to change this?

Reply:
TBH the the use of block copying for primitive arrays is my least favorite part of this library. I'd be keen on removing it completely in future releases.

Reply:
Buffer.BlockCopy works with ints so can't handle arrays of more than 2GB

Reply:
> I'd be keen on removing it completely in future releases.

And replacing it with what?

Reply:
We could branch based on the array size and use blockcopy < 2GB and pinning and Marshal.Copy >= 2GB.

Reply:
> And replacing it with what?

Pointwise copy. Serialization output that depends on machine endianness makes me nervous.